subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftBasePlugin.java

```diff
@@ -161,7 +160,7 @@ public String call() {
                     final CreateSwiftBundle bundle = tasks.create(names.getTaskName("bundleSwift"), CreateSwiftBundle.class);
                     bundle.getExecutableFile().set(link.getBinaryFile());
                     bundle.getInformationFile().set(((SwiftBundle) binary).getInformationPropertyList());
-                    Provider<Directory> bundleLocation = buildDirectory.dir(providers.provider(new Callable<String>() {
+                    Provider<RegularFile> bundleLocation = buildDirectory.file(providers.provider(new Callable<String>() {
```

Comment:
This is still used as an output dir below: `bundle.getOutputDir().set(bundleLocation)`

Is it supposed to be a file or a directory?

Reply:
I'm quite confused by Apple bundle idea. They are directories but should be treated as files (single unit). I just don't know how to model them. If I model them as a `Directory`, the logic in the Xcode plugins become complicated as it expects `Provider<RegularFile>` for the product location which is mapped to a file in Xcode. There isn't a way to convert a directory to a file to get the directory location as a lazy value. My thinking was that since Apple wants users to treat bundle as a single unit, let's assume it's a file. I agree we should change the getter to reflect this (probably use `getOutputFile()` or `getOutputBundle()`). What do you think?